public/js/reducers/pause.js

```diff
@@ -66,6 +67,21 @@ function update(state = initialState, action, emit) {
     case constants.PAUSE_ON_EXCEPTIONS:
       const toggle = action.toggle;
       return state.set("shouldPauseOnExceptions", toggle);
+
+    case constants.ADD_EXPRESSION:
+      return state.setIn(["expressions", action.id],
+        { id: action.id,
+          expression: action.expression,
+          value: action.value });
+
+    case constants.EVALUATE_EXPRESSION:
+      if (action.status === "done") {
+        return state.setIn(["expressions", action.id],
+          { id: action.id,
```

Comment:
We could probably pull these watch expressions into their own reducer, as the whole notion of "pause" state is too broad in my opinion. But that's something that can be done later.